mongodbatlas/data_source_mongodbatlas_privatelink_endpoint_service_adl.go

```diff
@@ -0,0 +1,81 @@
+package mongodbatlas
+
+import (
+	"context"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceMongoDBAtlasPrivateLinkEndpointServiceADL() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceMongoDBAtlasPrivateLinkEndpointServiceADLRead,
+		Schema: map[string]*schema.Schema{
+			"project_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"endpoint_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"provider_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"comment": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceMongoDBAtlasPrivateLinkEndpointServiceADLRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	// Get client connection.
+	conn := meta.(*MongoDBClient).Atlas
+	projectID := d.Get("project_id").(string)
+	endpointID := d.Get("endpoint_id").(string)
+
+	privateLinkResponse, _, err := conn.DataLakes.GetPrivateLinkEndpoint(ctx, projectID, endpointID)
+	if err != nil {
+		// case 404
+		// deleted in the backend case
+		reset := strings.Contains(err.Error(), "404") && !d.IsNewResource()
```

Comment:
[q] Why did you add `&& !d.IsNewResource()`? why getting 404 is not enough to remove the resource?

Reply:
this was added to avoid eventual inconsistency but in this case is not necessary